k2/csrc/fsa_utils_test.cu

```diff
@@ -143,6 +178,53 @@ TEST(FsaFromString, Transducer) {
   }
 }
 
+TEST(FsaFromString, OpenFstTransducer) {
+  // src_state dst_state label aux_label cost
+  std::string s = R"(0 1 2 22  -1.2
+    0 2  10 100 -2.2
+    1 3  3  33  -3.2
+    1 6  4  16  -4.2
+    2 6  5  26  -5.2
+    2 4  2  22  -6.2
+    3 6  7  36  -7.2
+    5 7  1  50  -8.2
+    6 -1.2
```

Comment:
If I remember correctly, OpenFst does not require that the state number
of a final state has to be larger than that of a non-final state.

I'm afraid the current implementation cannot be used to read an FST in OpenFst format
as it will abort if the first column is not in non-decreasing order.

Reply:
we'd add a new super-final state regardless, i.e. the final-state becomes an arc like
6 8 -1   0  -1.2

Reply:
yes, I saw that. The problem is that if you switch
```
    6 -1.2
    7 -2.3
```
to
```
    7 -2.3
    6 -1.2
```

I believe the program will crash.  Since @chenguoguo is using `std::max` to compute the max state, he
is not assuming that the first column is in non-decreasing order.

Reply:
I think the code is adding
```cpp
6 8 -1 -1 -1.2
```

instead of

```cpp
6 8 -1 0 -1.2
```